.github/workflows/reusable-on-code-change.yml

```diff
@@ -0,0 +1,198 @@
+name: Reusable Workflow (On Code Change)
+on:
+  workflow_call:
+    inputs:
+        static_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        build_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        sonar_analysis_instance_type:
+            type: string
+            default: "ubuntu-latest"
+        integration_tests_instance_type:
+            type: string
+            default: "self-hosted"
+        integration_tests_standalone_execute:
+            type: boolean
+            default: false
+        integration_tests_cluster_execute:
+            type: boolean
+            default: false
+        static_analysis_node_version:
+            type: number
+            default: 18
+        mvn_settings_filepath:
+            type: string
+            default: ".github/maven.settings.xml"
+        static_analysis_auditci_level:
+            type: string
+            default: "moderate"
+        module_id:
+            type: string
+            required: true
+        integration_tests_testrail_project:
+            type: string
+            default: ""
+        integration_tests_pagerduty_skip_notification:
+            type: boolean
+            default: true
+        integration_tests_provisioning_manifest:
+            type: string
+            default: ""
+        integration_tests_should_use_build_artifacts:
+            type: boolean
+            default: false
+        integration_tests_artifact_prefix:
+            type: string
+            default: "tests"
+        integration_tests_jahia_image:
+            type: string
+            default: ""
+        integration_tests_pagerduty_incident_service:
+            type: string
+            default: ""
+        integration_tests_timeout:
+            type: number
+            default: 65
+        integration_tests_timeout_step:
+            type: number
+            default: 50
+        integration_tests_profile:
+            type: string
+            default: ""
+        integration_tests_elasticsearch_image:
+            type: string
+            default: ""
+        integration_tests_jcustomer_image:
+            type: string
+            default: ""
+        integration_tests_should_skip_testrail:
+            type: boolean
+            default: false
+        integration_tests_should_skip_pagerduty:
+            type: boolean
+            default: false
+        sonar_analysis_primary_release_branch:
+            type: string
+            default: "master"
+        module_branch:
+            required: true
+            type: string
+        build_container_image:
```

Comment:
Can you organize the inputs by job (in the same order than the job order) ?
 - general inputs
 - build_instance_type
 - build_container_image